tests/integration/api/mutations/captureOrderPayments/captureOrderPayments.test.js

```diff
@@ -0,0 +1,145 @@
+import encodeOpaqueId from "@reactioncommerce/api-utils/encodeOpaqueId.js";
+import importAsString from "@reactioncommerce/api-utils/importAsString.js";
+import Factory from "/tests/util/factory.js";
+import TestApp from "/tests/util/TestApp.js";
+
+const captureOrderPayments = importAsString("./captureOrderPayments.graphql");
+
+jest.setTimeout(300000);
+
+const shopId = "123";
+const opaqueShopId = encodeOpaqueId("reaction/shop", shopId); // reaction/shop:123
+const orderId = "456";
+const opaqueOrderId = encodeOpaqueId("reaction/order", orderId);
+const paymentId = "789";
+const opaquePaymentId = encodeOpaqueId("reaction/payment", paymentId);
+const productId = "product1";
+const productVariantId = "variant1";
+
+const shopName = "Test Shop";
+
+const mockAdminAccount = Factory.Account.makeOne({
+  roles: {
+    [shopId]: ["admin", "orders"]
+  },
+  shopId
+});
+
+const product = Factory.Product.makeOne({
+  _id: productId,
+  ancestors: [],
+  handle: "test-product",
+  isDeleted: false,
+  isVisible: true,
+  shopId,
+  type: "simple"
+});
+
+const variant = Factory.Product.makeOne({
+  _id: productVariantId,
+  ancestors: [productId],
+  attributeLabel: "Variant",
+  isDeleted: false,
+  isVisible: true,
+  shopId,
+  type: "variant"
+});
+
+let testApp;
+let captureOrderPaymentsMutation;
+
+beforeAll(async () => {
+  testApp = new TestApp();
+  await testApp.start();
+
+  await testApp.insertPrimaryShop({ _id: shopId, name: shopName });
+  await testApp.collections.Products.insertOne(product);
+  await testApp.collections.Products.insertOne(variant);
+  await testApp.publishProducts([productId]);
+
+  await testApp.createUserAndAccount(mockAdminAccount);
+  captureOrderPaymentsMutation = testApp.mutate(captureOrderPayments);
+});
+
+afterAll(async () => {
+  await testApp.collections.Accounts.deleteMany({});
+  await testApp.collections.users.deleteMany({});
+  await testApp.collections.Shops.deleteMany({});
+  await testApp.collections.Products.deleteMany({});
+  await testApp.stop();
+});
+
+test("an anonymous user should not be able to capture order payments", async () => {
+  try {
+    await captureOrderPaymentsMutation({
+      input: {
+        orderId: opaqueOrderId,
+        paymentIds: [opaquePaymentId],
+        shopId: opaqueShopId
+      }
+    });
+  } catch (error) {
+    expect(error).toMatchSnapshot();
+    return;
+  }
+});
+
+test("an admin user should be able to capture order payments", async () => {
+  let result;
+  await testApp.setLoggedInUser(mockAdminAccount);
+
+  // Place an order
+  const orderItem = Factory.OrderItem.makeOne({
```

Comment:
These factories should be in the `beforeAll` or `beforeEach` with cleanup on `afterEach` so that we ensure the anonymous user has an existing order that can be captured, and their denial is purely permission-based.

Reply:
moved